interface.go

```diff
@@ -369,25 +363,27 @@ func (c *Client) DownloadURL(u string, output io.Writer) error {
 	// let's seek 0 bytes from the end and determine the new offset which is the
 	// file's size
 	if s, ok := output.(io.Seeker); ok {
-		size, err := s.Seek(0, io.SeekEnd)
+		var size int64
+		size, err = s.Seek(0, io.SeekEnd)
 		if err == nil && size != 0 {
 			return ErrBadOutputWriter
 		}
 	}
 
-	request := newRequest(u, "GET", &http.Header{})
+	r := newRequest(u, "GET", &http.Header{})
 
 	var redirectBuf bytes.Buffer
 
-	cs, _, err := c.agent.run(request, nil, c.chunkSize, &redirectBuf, false)
+	var cs callSummary
+	cs, _, err = c.agent.run(r, nil, c.chunkSize, &redirectBuf, false)
 
 	var storageType string
 	if cs.ResponseHeader != nil {
 		storageType = cs.ResponseHeader.Get("x-taskcluster-artifact-storage-type")
 	}
 
 	if err != nil && storageType != "error" {
-		logger.Printf("%s\n%s", cs, redirectBuf.String())
+		logger.Printf("%s\n%v", cs, redirectBuf.String())
```

Comment:
wouldn't we either want to drop the `.String()` or keep the `%s` formatter?

Reply:
So that's what I thought at first, but after testing, I discovered it didn't work (see [here](https://play.golang.org/p/XTLyg_cRMaN)). This was also a surprise to me!

However, perhaps option 6 (the go playground link above) is cleanest?

Reply:
yes!  that looks good.  I just tested with an emoji with zero width joiners (👪) and both `%v` and `%s` work as expected, which addresses my only concern with v over s.

Let's go with option 6 :)